lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -23,11 +23,12 @@ def list(rgroup = configuration.resource_group)
         JSON.parse(response)['value'].map { |hash| model_class.new(hash) }
       end
 
-      def list_all
+      def list_all(filter = {})
```

Comment:
Eh, I tend to think of a filter as a single thing with multiple properties. Plus, it's shorter. ;)

I added some comments.